Various updates, clarifications, and fixes to the BeeWare tutorial.

kattni opened the pull request:
<!--- Describe your changes in detail -->
- Tutorial 1: Updates the list of details requested by Briefcase to match the names listed when running `briefcase new`. https://github.com/beeware/beeware/pull/524/commits/8a2613ab3a716d461f65a0041dede5d4f941fd64
- Tutorial 2: Clarify and simplify explanation of `__main__.py`. https://github.com/beeware/beeware/pull/524/commits/b3360b5b99aed50496a1572a4d85b80934500430
- Tutorial 2: Add stronger wording about the usage of `-r` when dealing with updated dependencies. https://github.com/beeware/beeware/pull/524/commits/d2ef1b5a0e7db97f6c983e0853a571fba0e4cb77
- Tutorial 2: Add note about halting the app using CTRL+C. https://github.com/beeware/beeware/pull/524/commits/14c3c955ecd3cfbdabcefc835ff0614224dd47c4
- Tutorial 3: Add note about halting the app using CTRL+C. https://github.com/beeware/beeware/pull/524/commits/14c3c955ecd3cfbdabcefc835ff0614224dd47c4
- Tutorial 5 - Android: Add note about not accepting the default option to review necessary licenses, which is "No". https://github.com/beeware/beeware/pull/524/commits/b6cedc69ebca08c2081b0da18426c397b10b8fc3
- Tutorial 5 - Android: Add note about CTRL+C halting the log messages but not closing the emulator. https://github.com/beeware/beeware/pull/524/commits/dc60f362d784fb19b340a259da206cc771f03dba
- Tutorial 5 - iOS: Add note about CTRL+C halting the log messages but not closing the simulator. https://github.com/beeware/beeware/pull/524/commits/93d9481775f78b6c56b3392f80bf44fe45d8dcfc
- Tutorial 5 - iOS: Update wording about XCode component and simulator installation. https://github.com/beeware/beeware/pull/524/commits/c7ad5ee7d89df4c1506161c2db5864d5f0575e20
- Tutorial 7: Update third-party package process to use `faker`, which is not used by Briefcase, as `httpx` is. https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd
- Tutorial 7: Update explanation of mobile and web third-party package limitations. https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd
- Tutorial 7: Update macOS screenshots. https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd
- Tutorial 7: Remove references to and explanation of `httpx`. https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd
- Tutorial 8: Add `httpx` explanation and changes, reword for continuity. https://github.com/beeware/beeware/pull/524/commits/1bd21207cb76647f60e6a1ede9c58ae00a699d31
- Going further - Using the camera: Fix typo in first paragraph. https://github.com/beeware/beeware/pull/524/commits/f9fd0290f92944a068466e467a3e4231f6cbdf08
- Going further - Using the camera: Removed reference to the "highlight color" being yellow in the updated code example, because it is a lighter grey in dark mode, and may be confusing. Alternatively, I can update it to say "yellow or grey". https://github.com/beeware/beeware/pull/524/commits/0e8749dc6b68da40f2942864a2404255e748bf90
- Going further - Using the camera: Updated example code to use `margin` instead of `padding`, as `padding` results in the following warning: `<string>:30: DeprecationWarning: Pack.padding is deprecated. Use Pack.margin instead.`.  This assumes that `margin` is a drop-in replacement for `padding`. https://github.com/beeware/beeware/pull/524/commits/06bc38a75560446424cd2cc22f4adea325ae9603
- Contributing: Update `pip` command to latest suggested version (as far as I understand it; please correct me if I'm wrong). https://github.com/beeware/beeware/pull/524/commits/f83246bb6acdc4eb89607dd009d2e1ca907fc672
- Tutorial 8: Rework per discussion below. https://github.com/beeware/beeware/pull/524/commits/552c345b0f7b0b79ba687f3bd2066714cf48eeb8
- Tutorial 8: Add updated screenshot for macOS. https://github.com/beeware/beeware/pull/524/commits/552c345b0f7b0b79ba687f3bd2066714cf48eeb8
<!--- What problem does this change solve? -->
<!--- If this PR relates to an issue, include Refs #XXX or Fixes #XXX -->

## PR Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

kattni commented:
@freakboy3742 I need the following screenshots from you, before this PR is merged:
* Linux:
  * Updated dialog box from the app code in this commit: https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd (located at docs/tutorial/images/linux/tutorial-7.png)
* Windows:
  * Updated dialog box from the app code in this commit: https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd (located at docs/tutorial/images/windows/tutorial-7.png)
  * Updated crash dialog box from the app code in this commit: https://github.com/beeware/beeware/pull/524/commits/17217ddd46865093916a904c8915871be43beedd (located at docs/tutorial/images/windows/tutorial-7-crash.png)

kattni commented:
I have thoughts on the [asynchronous programming section](https://docs.beeware.org/en/latest/tutorial/tutorial-8.html#asynchronous-programming). I think there is room to make this section more accessible, however, I don't know whether its accessibility is a concern of you all to begin with. I brought this up on Discord, and @rmartin16 responded. The suggestion was to link to existing documentation such as [this](https://realpython.com/async-io-python/) or [this](https://superfastpython.com/python-asyncio/) (though I feel the latter may be overwhelming and a less ideal choice). My concern is that linking out often means losing folks, but I also understand the desire to avoid duplicating work. I wanted to document this here in case there are any further opinions. I did not make any changes to that section at this time.

kattni commented:
This sentence in the Contributing section of the tutorial is misleading and may be contributing to confusion:
```
To contribute to BeeWare itself, not the docs, please review the repo README.
```
It is my understanding that BeeWare is the name for the suite of tools, not something in itself that even can be "contributed to", however that potential confusion source will soon be somewhat alleviated through the process currently being discussed [here](https://github.com/beeware/beeware/discussions/522). That said, the wording in the Contributing section will need to be reworded at that point, as the tutorial will be living in a dedicated repo, and will therefore not need this distinction made. 

I can update it now if that would be easier.

The README is confusing as well in the "BeeWare as s thing" sense. As, the repo will remain in existence, I think it is important to attempt to alleviate the potential for further confusion through an update to the README. That should happen in a subsequent PR.

kattni commented:
@freakboy3742 This is ready for review, at your convenience.

freakboy3742 reviewed (changes requested):
These updates all looks great. I've made a couple of minor tweaks; and added the missing screenshots for Windows and Linux.

As I see it, there's 2 things standing between this and merging.

1. A review of my edits - how many embarrassing typos have I made? :-P
2. The transition from Tutorial 7 to Tutorial 8 still needs a little work

In the "pre-faker" version, tutorial 7 used httpx in synchronous mode, which had the side effect that if your network connection wasn't lightning fast, you'd see a beach ball - which provided the transition into discussing event loops and why you needed to use the async mode for httpx. 

However, with Faker replacing httpx in T7, that no longer happens - so the "may have noticed the app beach balling" reference at the end of T7 and start of T8 doesn't make sense any more. 

I can think of three possible fixes:
1. We reframe T8 to be "how do you keep a GUI responsive?", have the discussion about GUI loops, and then run straight into the async httpx usage.
2. We take the "synchronous httpx" parts of the old T7 and add them to the start of T8, show that it locks up, then discuss GUI loops, and make the async change
3. We add a new section to the start of T8 that delibrately makes a "slow API" change - e.g., calling "time.sleep(5)" in the middle of `say_hello()`, then discuss GUI loops, and replace the "slow API" with "async httpx".

I'm open to other suggestions, though.

kattni commented:
> 1. A review of my edits - how many embarrassing typos have I made? :-P

Reviewed; your edits are solid. I do not have any suggested changes.

> 2. The transition from Tutorial 7 to Tutorial 8 still needs a little work

Agreed. I did the best I could with what was there. As I don't have a good grasp on the situation, it was basically me moving info around, with only a vague understanding of the concept as a whole.

> 2. We take the "synchronous httpx" parts of the old T7 and add them to the start of T8, show that it locks up, then discuss GUI loops, and make the async change

I think 2 is the best option of the three; it provides the "real" scenario the user is trying to fix, and transitions into the fix. The first one is the second best. however, I feel that experiencing the issue before trying to fix it is a better option. The third one feels like it would be incredibly stilted, and we still have to update the code for `httpx`, so in the case of this option, you would have to then remove the `time.sleep()` and add the `httpx` payload. 

I suggest going with your second idea.

freakboy3742 commented:
> I suggest going with your second idea.

Your analysis matches my thinking, so let's go with option 2.

kattni commented:
On it. I'll do what I can, but I have a feeling you're going to need to clean it up before we can land this.

kattni commented:
@freakboy3742 I need the following screenshots from you, before this PR is merged. See the image at `docs/tutorial/images/macos/tutorial-8.png` for the expected content.

- Linux:
    - Updated dialog box from the first instance of app code in this commit: https://github.com/beeware/beeware/pull/524/commits/552c345b0f7b0b79ba687f3bd2066714cf48eeb8 (to be added at docs/tutorial/images/linux/tutorial-8.png)
- Windows:
    - Updated dialog box from the first instance of app code in this commit: https://github.com/beeware/beeware/pull/524/commits/552c345b0f7b0b79ba687f3bd2066714cf48eeb8 (to be added at docs/tutorial/images/windows/tutorial-8.png)

kattni commented:
RtD is failing because the now-Tutorial-8 images aren't available yet. Should resolve itself once they are added.

freakboy3742 reviewed (approved):
This looks great, and I've added in the missing screenshots and lead-in intro text. 

If I can get a check that I haven't fallen over my own feet with that introduction, I'll merge.

kattni commented:
Looks good other than the one correction above.

kattni commented:
Reads well. Looks good to me.
